openbb_platform/obbject_extensions/charting/openbb_charting/__init__.py

```diff
@@ -0,0 +1,157 @@
+
+from openbb_core.app.model.charts.chart import Chart
+from openbb_core.app.model.extension import Extension
+
+from openbb_charting import charting_router
+from openbb_charting.core.to_chart import ChartIndicators, OpenBBFigure, to_chart
+from openbb_charting.utils.helpers import get_charting_functions
+
+ext = Extension(name="charting")
+
+
+@ext.obbject_accessor
+class Charting:
+    """
+    Charting extension.
+
+    Methods
+    -------
+    show
+        Display chart and save it to the OBBject.
+    to_chart
+        Returns the plotly json representation of the chart.
+    functions
+        Returns a list of Platform commands with charting functions.
+    indicators
+        Returns a list of the available indicators to use with the `to_chart` method.
+    """
+
+    def __init__(self, obbject):
+        """Initialize Charting extension."""
+        # pylint: disable=import-outside-toplevel
+        from openbb_core.app.model.charts.charting_settings import ChartingSettings
+        from openbb_core.app.model.obbject import OBBject
+
+        self._obbject: OBBject = obbject
+        self._charting_settings = ChartingSettings(
+            user_settings=self._obbject._user_settings,
+            system_settings=self._obbject._system_settings,
+        )
+        self._handle_backend()
+
+    @classmethod
+    def indicators(cls):
+        """Returns a list of the available indicators."""
+        return ChartIndicators.get_available_indicators()
+
+    @classmethod
+    def functions(cls):
+        """Returns a list of the available functions."""
+        return get_charting_functions()
+
+    def _handle_backend(self):
+        # pylint: disable=import-outside-toplevel
+        from openbb_charting.core.backend import create_backend, get_backend
+
+        create_backend(self._charting_settings)
+        get_backend().start(debug=self._charting_settings.debug_mode)
+
+    @staticmethod
+    def _get_chart_function(route: str) -> Callable:
+        """Given a route, it returns the chart function. The module must contain the given route."""
+        adjusted_route = route.replace("/", "_")[1:]
+        return getattr(charting_router, adjusted_route)
+
+    def show(self, render: bool = True, **kwargs):
+        """Display chart and save it to the OBBject."""
+        charting_function = self._get_chart_function(
+            self._obbject._route  # pylint: disable=protected-access
+        )
+        kwargs["obbject_item"] = self._obbject.results
+        kwargs["charting_settings"] = self._charting_settings
+        kwargs["standard_params"] = (
+            self._obbject._standard_params.__dict__  # pylint: disable=protected-access
+        )
+
+        fig, content = charting_function(**kwargs)
+        self._obbject.chart = Chart(
+            fig=fig, content=content, format=charting_router.CHART_FORMAT
+        )
+        if render:
+            fig.show()
+
+    def to_chart(
+        self,
```

Comment:
In `to_chart`, and `show`, I'm looking for an entry point that can take `List[Data]` and some variants.  A data param, something like: `data: Optional[Union[Data, List[Data], Dict, List[Dict], DataFrame, Series]] = None`, where `data` would override `self` before it created the figure, when supplied.

```
data = obb.equity.price.historical("AAPL", start_date="2000-01-01")
data.charting.to_chart(data = data.to_df().loc["2022-01-01":])
```

Reply:
I like this idea, this is something we already have in the data processing commands so it's really low effort to implement - will do it on this PR.